src/wallet/bdk.rs

```diff
@@ -1363,6 +1451,142 @@ impl Wallet {
+) -> Result<(ExtendedDescriptor, KeyMap), DescriptorError> {
+    let (descriptor, keymap) = inner.into_wallet_descriptor(secp, network)?;
+
+    // Ensure the keys don't contain any hardened derivation steps or hardened wildcards
+    let descriptor_contains_hardened_steps = descriptor.for_any_key(|k| {
+        if let DescriptorPublicKey::XPub(DescriptorXKey {
+            derivation_path,
+            wildcard,
+            ..
+        }) = k
+        {
+            return *wildcard == Wildcard::Hardened
+                || derivation_path
+                    .into_iter()
+                    .any(bdk_wallet::bitcoin::bip32::ChildNumber::is_hardened);
+        }
+
+        false
+    });
+    if descriptor_contains_hardened_steps {
+        return Err(DescriptorError::HardenedDerivationXpub);
+    }
+
+    if descriptor.is_multipath() {
+        return Err(DescriptorError::MultiPath);
+    }
+
+    // Run miniscript's sanity check, which will look for duplicated keys and other potential
+    // issues
+    descriptor.sanity_check()?;
+
+    Ok((descriptor, keymap))
+}
+
+impl Wallet {
+    fn create_wallet_descriptors(&self) -> Result<HashMap<KeychainKind, String>, anyhow::Error> {
+        // create master_key from seed
+
+        let secp = Secp256k1::new();
+
+        // create Xpriv from HARDENDED_DERIVATION path(till account no):
+        let wallet_xpriv = self
+            .store
+            .master_key
+            .derive_priv(
+                &secp,
+                &DerivationPath::from_str(HARDENDED_DERIVATION).unwrap(),
+            )
+            .unwrap();
+
+        // create descriptors for each keychain -> get hashmap.
+        // Note:
+        // created descriptor will contain xpriv not xpub
+
+        let mut kind_descriptor_map = HashMap::new();
+        //  let descriptors =
+        for keychain in KeychainKind::iterator() {
+            let descriptor_without_checksum = match *keychain {
+                KeychainKind::External | KeychainKind::Internal => {
+                    format!("wpkh({}/{}/*)", wallet_xpriv, keychain.index_num())
+                }
+                KeychainKind::Fidelity | KeychainKind::SwapCoin | KeychainKind::Contract => todo!(),
+            };
+            let descriptor = format!(
+                "{}#{}",
+                descriptor_without_checksum,
+                calc_checksum(&descriptor_without_checksum).unwrap()
+            );
+            // TODO: will always return None
+            kind_descriptor_map.insert(*keychain, descriptor);
+        }
+
+        Ok(kind_descriptor_map)
+    }
+}
+
+//_____________________________________WALLET/SIGNER______________________________________________
+impl Wallet {
+    fn create_signers<E>(
+        index: &mut KeychainTxOutIndex<KeychainKind>,
+        secp: &Secp256k1<All>,
+        kind_descriptor_map: HashMap<KeychainKind, E>,
+        network: Network,
+    ) -> Result<HashMap<KeychainKind, Arc<SignersContainer>>, DescriptorError>
```

Comment:
A function like this needs unit testing. Even though it might look okay in code, there could be tiny bugs. So let's ensure we are writing what we need to write with just a small unit test along with each function.

Make a random seed, derive xpubs from them, make the signer, and assert its inner data are as expected.